### PR TITLE
ADD yaml merging of all fields including daemon.json conf

### PR DESCRIPTION
**Pull request by mzaneri**

### CODE REVIEW

File: kurl_util/cmd/installermerge/main.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+		}
+
+		if key == "daemonConfig" {
+			mergedDockerConfig, _ := mergeDockerConfigData([]byte(oldVal.(string)), []byte(newVal.(string)))
+			mergedConfig[key] = string(mergedDockerConfig)
+			continue
+		}
+
+		oldValMap, isOldMap := oldVal.(map[interface{}]interface{})
+		newValMap, isNewMap := newVal.(map[interface{}]interface{})
```
Comment (divolgin) [inline kurl_util/cmd/installermerge/main.go:77]: Do you ever hit this case?  YAML and JSON can't have keys that aren't strings.

File: kurl_util/cmd/installermerge/main.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+		return nil, errors.Wrap(err, "failed to marshal merged config")
+	}
+
+	return mergedconfigdata, nil
+}
+
+func main() {
+	kurlscheme.AddToScheme(scheme.Scheme)
+
+	oldInstaller, _ := getInstallerConfigFromYaml("old.yaml")
```
Comment (divolgin) [inline kurl_util/cmd/installermerge/main.go:228]: Shouldn't these files come from command line arguments?  And also, error handling?

Comment (mzaneri) [reply kurl_util/cmd/installermerge/main.go:228]: the mergeConfigYamlData was really what were looking at, that will be done before its read for use in kurl

File: kurl_util/cmd/installermerge/main.go (hunk 1 above)
Comment (mzaneri) [reply kurl_util/cmd/installermerge/main.go:77]: yes, this is a quirk of YAML unmarshalling

**Pull request merged**